### DISCUSSION TITLE
[Request] Modify All-Day Event Free/Busy Availability to be "Free" for All-Day events

### DISCUSSION THREAD

**Discussion by mtemple** [derekantrican/GAS-ICS-Sync#423] [Q&A] [answered]
What is the best way to modify the script so that when it updates new All-Day events (duration = 24hrs), the all day event is listed as "Free" instead of "Busy". Google's default all-day value is to set it as Free, but the calendar I'm trying to subscribe to for some reason creates the ALL-DAY events as default BUSY.

Would this also work so that previous All-Day events that are "Free" that later are scheduled with a time (no longer all-day) are changed/reverted back to "Busy"?

**Comment by jonas0b1011001** [comment, marked as answer]
> Google's default all-day value is to set it as Free

I don't think that's true, the [API docs](https://developers.google.com/calendar/api/v3/reference/events) list busy as standard value.

---
I'd suggest changing
https://github.com/derekantrican/GAS-ICS-Sync/blob/f5c8eb5ee5b364f1c03175627163c2fb6c724b44/Helpers.gs#L479-L483
to
```
  if (icalEvent.startDate.isDate){
    newEvent.transparency = "transparent";
  }
  else if (event.hasProperty('transp')){
    var transparency = event.getFirstPropertyValue('transp').toString().toLowerCase();
    if(["opaque", "transparent"].indexOf(transparency) > -1)
      newEvent.transparency = transparency;
  }
```
After the edit, force a resync by syncing to a different target calendar.

**Reply by mtemple** [reply]
Thanks - I'll try that out and see what happens / report back!

This is why I thought the all-day value is default set to Free. When making a new all-day event directly in gCal [it shows the event's transparency value as "Free"](https://share.cleanshot.com/vrQCVdwJ)

**Comment by jonas0b1011001** [comment]
I assume this is resolved / no longer relevant. Closing.
